ruby/example_code/cloudtrail/aws-ruby-sdk-cloudtrail-example-describe-trails.rb

```diff
@@ -21,19 +21,66 @@
 # OF ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-require 'aws-sdk-cloudtrail'  # v2: require 'aws-sdk'
+require 'aws-sdk-cloudtrail'
 
-# Create client in us-west-2
-client = Aws::CloudTrail::Client.new(region: 'us-west-2')
+# Lists information about existing trails in AWS CloudTrail.
+class ListTrailsExample
+  # Initialize an instance of ListTrailsExample, creating a client for
+  # AWS CloudTrail (unless already provided during initialization).
+  #
+  # (The following comments express documentation about this function in YARD 
+  # format by using @ symbols.)
+  #
+  # @param [Hash] opts ({}) A hash of an API client for CloudTrail.
+  # @option [Aws::CloudTrail::Client] :cloudtrail_client
+  #  (Aws::CloudTrail::Client)
+  def initialize(opts = {})
+    # This CloudTrail API client is used to list the CloudTrail resources.
+    @cloudtrail = opts[:cloudtrail_client] || Aws::CloudTrail::Client.new
+  end
 
-resp = client.describe_trails({})
+  # Lists selected information about existing trails in AWS CloudTrail.
+  def list_trails()
+    resp = @cloudtrail.describe_trails({})
+    
+    puts
+    puts "Found #{resp.trail_list.count} trail(s):"
+    puts
+    
+    resp.trail_list.each do |trail|
+      puts 'Name:                  ' + trail.name

```

Comment:
Minor: In other examples, you were using "#{string} interpolation" (preferred). You can also use `\t` in double quoted strings for tabs instead of counting spaces.